src/httpserver.cpp

```diff
@@ -0,0 +1,103 @@
+#ifdef PANDA3DS_ENABLE_HTTP_SERVER
+#include "httpserver.hpp"
+
+#include <vector>
+#include <map>
+#include <thread>
+#include <fstream>
+#include <string>
+
+#include "httplib.h"
+#include "services/hid.hpp"
+
+u32 stringToKey(const std::string& key_name) {
+	namespace Keys = HID::Keys;
+	static std::map<std::string, u32> keyMap = {
+		{"A", Keys::A},
+		{"B", Keys::B},
+		{"Select", Keys::Select},
+		{"Start", Keys::Start},
+		{"Right", Keys::Right},
+		{"Left", Keys::Left},
+		{"Up", Keys::Up},
+		{"Down", Keys::Down},
+		{"R", Keys::R},
+		{"L", Keys::L},
+		{"X", Keys::X},
+		{"Y", Keys::Y},
+		{"CirclePadRight", Keys::CirclePadRight},
+		{"CirclePadLeft", Keys::CirclePadLeft},
+		{"CirclePadUp", Keys::CirclePadUp},
```

Comment:
The SkyEmu protocol supports floating point values as inputs, so maybe CirclePadX and CirclePadY might make more sense than, up/down/left/right.

Reply:
I'll change it in upcoming patch